tianshou/data/collector.py

```diff
@@ -20,17 +22,40 @@
     VectorReplayBuffer,
     to_numpy,
 )
+from tianshou.data.buffer.base import MalformedBufferError
 from tianshou.data.types import (
+    ActBatchProtocol,
+    DistBatchProtocol,
     ObsBatchProtocol,
     RolloutBatchProtocol,
 )
 from tianshou.env import BaseVectorEnv, DummyVectorEnv
 from tianshou.policy import BasePolicy
+from tianshou.policy.base import episode_mc_return_to_go
+from tianshou.utils.array import bisect_left, bisect_right
 from tianshou.utils.print import DataclassPPrintMixin
 from tianshou.utils.torch_utils import torch_train_mode
 
 log = logging.getLogger(__name__)
 
+_TArrLike = TypeVar("_TArrLike", bound="np.ndarray | torch.Tensor | Batch | None")
+
+
+def _get_start_stop_tuples_around_edges(
+    edges: Sequence[int],
+    start: int,
+    stop: int,
+) -> tuple[tuple[int, int], tuple[int, int]]:
+    """We assume that stop is smaller than start and that `edges` is a sorted array of integers.
+
+    Then it will return the two tuples containing (start, stop) where we go from start to the next edge,
+    and from the previous edge to stop.
+    :return: (start, upper_edge), (lower_edge, stop)
+    """
```

Comment:
Docstring is not sufficient to understand what this function is for.

Reply:
I'll add examples and explain why we need this

Reply:
I had to look at it forever to get it myself. Probably the result of some late night coding in the past.

I will remove this function, it shouldn't exist